pkg/xmtp/client.go

```diff
@@ -2,24 +2,35 @@ package xmtp
 
 import (
 	"context"
+	"crypto/tls"
 	"fmt"
 
 	v1 "github.com/xmtp/proto/go/message_api/v1"
 	"google.golang.org/grpc"
+	"google.golang.org/grpc/credentials"
 	"google.golang.org/grpc/credentials/insecure"
 )
 
-func newConn(ctx context.Context, apiAddress string) (*grpc.ClientConn, error) {
+func newConn(ctx context.Context, apiAddress string, useTls bool) (*grpc.ClientConn, error) {
 	dialAddr := fmt.Sprintf(apiAddress)
 	return grpc.DialContext(
 		ctx,
 		dialAddr,
-		grpc.WithTransportCredentials(insecure.NewCredentials()),
+		grpc.WithTransportCredentials(getCredentials(useTls)),
 	)
 }
 
-func NewClient(ctx context.Context, apiAddress string) (v1.MessageApiClient, error) {
-	conn, err := newConn(ctx, apiAddress)
+func getCredentials(useTls bool) credentials.TransportCredentials {
+	if useTls {
+		return credentials.NewTLS(&tls.Config{
+			InsecureSkipVerify: true,
```

Comment:
Why do we need this?

Reply:
Good call out. I double checked and we don't actually need it.